apps/client/src/app/export/export.component.ts

```diff
@@ -43,11 +46,41 @@ export class ExportComponent implements OnInit {
     private cookieService: CookieService,
     private githubService: GithubService,
     private fhirService: FhirService,
-    public configService: ConfigService) {
+    public configService: ConfigService,
+    public http: HttpClient) {
 
     this.options.implementationGuideId = this.cookieService.get(Globals.cookieKeys.exportLastImplementationGuideId + '_' + this.configService.fhirServer);
     this.options.responseFormat = <any>this.cookieService.get(Globals.cookieKeys.lastResponseFormat) || 'application/json';
     this.options.downloadOutput = true;
+    this.options.template = <any>this.cookieService.get(Globals.cookieKeys.lastExportTemplate) || this.options.template;
+    this.options.templateVersion = <any>this.cookieService.get(Globals.cookieKeys.lastExportTemplateVersion) || this.options.templateVersion;
+
+    this.templateChanged
```

Comment:
I suggest using the (change) attribute on the HTML element, instead of an EventEmitter().
You can create a method in the .ts file, like so:
```
  templateChanged() {
    this.templateVersions = [];
    // the remaining logic
  }
```

and then call it whenever the value of the input field changes using this:

```
<input type="text" class="form-control" [(ngModel)]="options.template" (change)="templateChanged()" />
```

Reply:
Actually, I see below that you're already using similar logic for `templateHasChanged()`
I'd suggest moving this logic inside `templateHasChanged()` so that you don't need the event emitter.